python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,781 @@
+        @param[in] convergenceMetric: Quality of fit metric for the matched templates of the input images.
+        @param[in] baseMask: Mask of the initial template coadd.
+        """
+        bboxGrow = afwGeom.Box2I(bbox)
+        bboxGrow.grow(self.config.bufferSize)
+        for model in dcrModels:
+            bboxGrow.clip(model.getBBox(afwImage.PARENT))
+        tempExpName = self.getTempExpDatasetName(self.warpType)
+        residualGeneratorList = []
+        weightList = []
+        convergeMask = afwImage.Mask.getPlaneBitMask(self.config.convergenceMaskPlanes)
+
+        for tempExpRef, imageScaler, altMaskSpans in zip(tempExpRefList, imageScalerList, altMaskList):
+            exposure = tempExpRef.get(tempExpName + "_sub", bbox=bboxGrow)
+            visitInfo = exposure.getInfo().getVisitInfo()
+            wcs = exposure.getInfo().getWcs()
+            maskedImage = exposure.getMaskedImage()
+            templateImage = self.buildMatchedTemplate(dcrModels, visitInfo, bboxGrow, wcs, mask=baseMask)
+            if exposure.getWcs().getPixelScale() != self.pixelScale:
+                self.log.warn("Incompatible pixel scale for %s %s", tempExpName, tempExpRef.dataId)
+            imageScaler.scaleMaskedImage(maskedImage)
+            mask = maskedImage.getMask()
+            if altMaskSpans is not None:
+                self.applyAltMaskPlanes(mask, altMaskSpans)
+
+            if self.config.removeMaskPlanes:
+                mask = maskedImage.getMask()
+                for maskPlane in self.config.removeMaskPlanes:
+                    try:
+                        mask &= ~mask.getPlaneBitMask(maskPlane)
+                    except Exception as e:
+                        self.log.warn("Unable to remove mask plane %s: %s", maskPlane, e.message)
+            maskedImage -= templateImage
+            obsWeight = self.calculateWeight(maskedImage, convergeMask)*1e3
+            if self.config.doAirmassWeight:
+                obsWeight *= visitInfo.getBoresightAirmass()
+            weightList.append(obsWeight)
+            residualGeneratorList.append(self.dcrResiduals(dcrModels, maskedImage, visitInfo, bboxGrow, wcs))
+
+        dcrSubModelOut = []
+        with self.timer("stack"):
+            for oldModel in dcrModels:
+                residualsList = [next(residualGenerator) for residualGenerator in residualGeneratorList]
+                residual = afwMath.statisticsStack(residualsList, statsFlags, statsCtrl, weightList,
+                                                   afwImage.Mask.getPlaneBitMask("CLIPPED"),
+                                                   afwImage.Mask.getPlaneBitMask("NO_DATA"))
+                residual.setXY0(bboxGrow.getBegin())
+                newModel = self.clampModel(residual, oldModel, bboxGrow,
+                                           useNonNegative=self.config.useNonNegative,
+                                           clamp=self.config.clampModel)
```

Comment:
Why do you need to pass config parameters that are already readily accessible to the method?